db.go

```diff
@@ -660,6 +666,31 @@ func (a *dbAppender) Rollback() error {
 	return g.Wait()
 }
 
+// Delete implements deletion of metrics.
+func (db *DB) Delete(mint, maxt int64, ms ...labels.Matcher) error {
+	db.cmtx.Lock()
+	defer db.cmtx.Unlock()
+
+	db.headmtx.RLock()
+	blocks := db.blocksForInterval(mint, maxt)
+	db.headmtx.RUnlock()
+
+	var g errgroup.Group
+
+	for _, b := range blocks {
+		f := func() error {
+			return b.Delete(mint, maxt, ms...)
+		}
+		g.Go(f)
```

Comment:
No need to instantiate `f`. Just pass the function directly to `g.Go`. It's a matter of test in the end of course.